sharktank/sharktank/types/layouts.py

```diff
@@ -598,14 +598,14 @@ def __init__(
         block_size: int = 32,
         use_fe8m0_scale: bool = True,
     ):
-        assert iterables_equal(
-            qs.shape[:-1], d.shape
-        ), "TODO: remove when this class is refactored to comply with BlockScaledLayout"
+        if len(qs.shape) == len(d.shape) + 1:
+            # Legacy scale format with no trailing singleton dimension to match qs.
+            # This is here to avoid breaking existing IRPA files.
+            d = d.unsqueeze(-1)
```

Comment:
Can you put a warning in the console or the logger to let people know they are using an irpa format that they should regenerate. Gives them warning ahead of time because we don't want the extra code to stick around forever.

Reply:
I added a warning.